src/devices_models/devices/thermal_generation.jl

```diff
@@ -60,191 +61,154 @@ get_variable_binary(::Union{ColdStartVariable, WarmStartVariable, HotStartVariab
+) #  -> Union{Nothing, NamedTuple{(:startup, :shutdown), Tuple{Float64, Float64}}}
+    (
+        min = 0.0,
+        max = PSY.get_active_power_limits(device).max -
+              PSY.get_active_power_limits(device).min,
     )
 end
 
-function custom_active_power_constraints!(
+"""
+Semicontinuous range constraints for unit commitment formulations
+"""
+function add_constraints!(
     container::OptimizationContainer,
-    ::IS.FlattenIteratorWrapper{T},
-    ::Type{<:ThermalDispatchNoMin},
-) where {T <: PSY.ThermalGen}
-    var_key = VariableKey(ActivePowerVariable, T)
-    variable = get_variable(container, var_key)
-    # If the variable was a lower bound != 0, not removing the LB can cause infeasibilities
-    for v in variable
-        if JuMP.has_lower_bound(v)
-            JuMP.set_lower_bound(v, 0.0)
-        end
-    end
+    T::Type{<:PowerVariableLimitsConstraint},
+    U::Type{<:VariableType},
+    devices::IS.FlattenIteratorWrapper{V},
+    model::DeviceModel{V, W},
+    X::Type{<:PM.AbstractPowerModel},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+) where {V <: PSY.ThermalGen, W <: AbstractThermalUnitCommitment}
+    add_semicontinuous_range_constraints!(container, T, U, devices, model, X, feedforward)
 end
 
 """
-This function adds the active power limits of generators. Constraint (17) & (18) from PGLIB
+Startup and shutdown active power limits for Compact Unit Commitment
 """
-function DeviceRangeConstraintSpec(
-    ::Type{<:ActivePowerVariableLimitsConstraint},
-    ::Type{PowerAboveMinimumVariable},
-    ::Type{T},
+function get_startup_shutdown_limits(
+    device,
+    ::Type{ActivePowerVariableLimitsConstraint},
     ::Type{<:ThermalMultiStartUnitCommitment},
-    ::Type{<:PM.AbstractPowerModel},
-    feedforward::Nothing,
-    use_parameters::Bool,
-) where {T <: PSY.ThermalMultiStart}
-    return DeviceRangeConstraintSpec(;
-        range_constraint_spec = RangeConstraintSpec(;
-            constraint_type = ActivePowerVariableLimitsConstraint(),
-            variable_type = PowerAboveMinimumVariable(),
-            limits_func = x -> (
-                min = 0.0,
-                max = PSY.get_active_power_limits(x).max -
-                      PSY.get_active_power_limits(x).min,
-            ),
-            bin_variable_types = [OnVariable(), StartVariable(), StopVariable()],
-            constraint_func = device_multistart_range!,
-            constraint_struct = DeviceMultiStartRangeConstraintsInfo,
-            lag_limits_func = PSY.get_power_trajectory,
-            component_type = T,
-        ),
+)
+    PSY.get_power_trajectory(device)
+end
+
+"""
+Min and Max active power limits for Compact Unit Commitment
+"""
+function get_min_max_limits(
+    device,
+    ::Type{ActivePowerVariableLimitsConstraint},
+    ::Type{<:AbstractCompactUnitCommitment},
+) #  -> Union{Nothing, NamedTuple{(:startup, :shutdown), Tuple{Float64, Float64}}}
+    (
+        min = PSY.get_active_power_limits(device).min,
```

Comment:
This should also be similar to ThermalMultiStart
```
min = 0,  max = PSY.get_active_power_limits(device).max -  PSY.get_active_power_limits(device).min
```

Reply:
Resolved